python/tank/api.py

```diff
@@ -520,6 +520,21 @@ def context_from_entity(self, entity_type, entity_id):
         """
         return context.from_entity(self, entity_type, entity_id)
 
+    def context_from_entity_dictionary(self, entity_dictionary):
+        """
+        Derives a context from a shotgun entity dictionary.  This will try to use any 
+        linked information available in the dictionary where possible but if it can't 
+        determine a valid context then it will fall back to 'context_from_entity' which 
+        may result in a Shotgun path cache query and be considerably slower.
```

Comment:
I think we need some examples to help illustrate this feature. If I am reading the code right, the following inputs are valid - is that right Alan?:

```
{type:Task, id: 1234}
{type:Task, id: 1234, project: {type: Project, id:12}}
{type:Task, id: 1234, name: Lighting}
{type:Shot, id: 123, name: foo, sg_sequence: {type: Sequence, id: 12, name: foo}, project: {type: Project, id: 12}}
```

It would also be cool with some specific examples to help users of this method understand when it is fast and slow - this is a method to help build high performance context lookups - what do I need to supply in order for these speedups to kick in? When is it using the path cache etc?